Code/Framework/AzFramework/AzFramework/Application/Application.h

```diff
@@ -51,14 +51,19 @@ namespace AzFramework
         class Implementation
         {
         public:
-            static Implementation* Create();
-
             virtual ~Implementation() = default;
             virtual void PumpSystemEventLoopOnce() = 0;
             virtual void PumpSystemEventLoopUntilEmpty() = 0;
             virtual void TerminateOnError(int errorCode) { exit(errorCode); }
         };
 
+        struct ImplementationFactory
+        {
+            AZ_TYPE_INFO(ImplementationFactory, "{D840FD45-97BC-40D6-A92B-C83B607EA9D5}");
+            virtual ~ImplementationFactory() = default;
+            virtual Implementation* Create() = 0;
```

Comment:
To use this class with `azrtti_cast`, the AZ_RTTI* macros would need to be used instead, so that appropriate virtual functions for casting to an ImplementationFactory to a derived classes are being added.